activerecord/lib/active_record/secure_token.rb

```diff
@@ -13,25 +13,27 @@ module ClassMethods
       #
       #   user = User.new
       #   user.save
-      #   user.token # => "pX27zsMN2ViQKta1bGfLmVJE"
-      #   user.auth_token # => "77TMHrHJFvFDwodq8w7Ev2m7"
+      #   user.token # => "NG91NSDQ3UXZFM4HK6T2KBZTS"
+      #   user.auth_token # => "7QQN2QYDIYTNXY9DNLPFTUY2P"
       #   user.regenerate_token # => true
+      #   user.token # => "5V23QSU9GNGS1QTMHDSZ28EKC"
       #   user.regenerate_auth_token # => true
+      #   user.auth_token # => "S338RREJ5MKWPDXLEHO9POU1Q"
       #
-      # <tt>SecureRandom::base58</tt> is used to generate the 24-character unique token, so collisions are highly unlikely.
+      # <tt>SecureRandom::random_number</tt> is used to generate the 25-character unique token, so collisions are highly unlikely.
       #
       # Note that it's still possible to generate a race condition in the database in the same way that
       # {validates_uniqueness_of}[rdoc-ref:Validations::ClassMethods#validates_uniqueness_of] can.
       # You're encouraged to add a unique index in the database to deal with this even more unlikely scenario.
       def has_secure_token(attribute = :token)
         # Load securerandom only when has_secure_token is used.
-        require 'active_support/core_ext/securerandom'
+        require 'securerandom'
         define_method("regenerate_#{attribute}") { update! attribute => self.class.generate_unique_secure_token }
         before_create { self.send("#{attribute}=", self.class.generate_unique_secure_token) unless self.send("#{attribute}?")}
       end
 
       def generate_unique_secure_token
-        SecureRandom.base58(24)
+        SecureRandom.random_number(1<<128).to_s(35).upcase.tr('0','Z').rjust(25,'Z')
```

Comment:
Moving the base35 implementation into this one usage means apps lose a convenient public way to generate a compact URL/QR/etc-safe token.

Perhaps it could be exposed as something like `SecureRandom.random_token(bits, alphabet: …)`